utils/nctl/sh/scenarios/itst02.sh

```diff
@@ -0,0 +1,200 @@
+    local LFB2=${2}
+
+    if [ "$LFB1" = "$LFB2" ]; then
+       log "error: $LFB1 = $LFB2, chain didn't progress."
+       exit 1
+    fi
+}
+
+function assert_chain_stalled() {
+    # Fucntion checks that the two remaining node's LFB checked
+    # n-seconds apart doesnt progress
+    log_step "ensuring chain stalled"
+    local SLEEP_TIME=${1}
+    local LFB_1_PRE=$(do_read_lfb_hash 1)
+    local LFB_2_PRE=$(do_read_lfb_hash 2)
+    log "Sleeping ${SLEEP_TIME}s..."
+    sleep $SLEEP_TIME
+    local LFB_1_POST=$(do_read_lfb_hash 1)
+    local LFB_2_POST=$(do_read_lfb_hash 2)
+
+    if [ "$LFB_1_PRE" != "$LFB_1_POST" ] && [ "$LFB_2_PRE" != "$LFB_2_POST" ]; then
+       log "Error: Chain progressed."
+       exit 1
+    fi
+
+    if [ "$LFB_1_POST" != "$LFB_2_POST" ]; then
+        log "Error: LFB mismatch on nodes"
+	exit 1
+    else
+        STALLED_LFB=$LFB_1_POST
+    fi
+}
+
+function do_await_genesis_era_to_complete() {
+    log_step "awaiting genesis era to complete"
+    while [ "$(get_chain_era)" -lt 1 ]; do
+        sleep 1.0
+    done
+}
+
+function do_read_lfb_hash() {
+    local NODE_ID=${1}
+    LFB_HASH=$(render_last_finalized_block_hash "$NODE_ID" | cut -f2 -d= | cut -f2 -d ' ')
+    echo "$LFB_HASH"
+}
+
+function do_stop_node() {
+    local NODE_ID=${1}
+    log_step "stopping node-$NODE_ID."
+    do_node_stop "$NODE_ID"
+    sleep 1
+}
+
+function do_start_node() {
+    local NODE_ID=${1}
+    log_step "starting node-$NODE_ID. Syncing from hash=${STALLED_LFB}"
+    do_node_start "$NODE_ID" "$STALLED_LFB"
+    sleep 1
+    if [ "$(do_node_status ${NODE_ID} | awk '{ print $2 }')" != "RUNNING" ]; then
+        log "ERROR: node-${NODE_ID} is not running"
+	exit 1
+    fi
+}
+
+function check_network_sync() {
+    local WAIT_TIME_SEC=0
+    log_step "check all nodes are in sync"
+    while [ "$WAIT_TIME_SEC" != "$SYNC_TIMEOUT_SEC" ]; do
+        if [ "$(do_read_lfb_hash '5')" = "$(do_read_lfb_hash '1')" ] && \
+		[ "$(do_read_lfb_hash '4')" = "$(do_read_lfb_hash '1')" ] && \
+		[ "$(do_read_lfb_hash '3')" = "$(do_read_lfb_hash '1')" ] && \
+		[ "$(do_read_lfb_hash '2')" = "$(do_read_lfb_hash '1')" ]; then
+	    log "all nodes in sync, proceeding..."
+	    break
+        fi
+
+        if [ "$WAIT_TIME_SEC" = "$SYNC_TIMEOUT_SEC" ]; then
+            log "ERROR: Failed to confirm network sync"
+            exit 1
+        fi
+        WAIT_TIME_SEC=$((WAIT_TIME_SEC + 1))
+        sleep 1
+    done
+}
+
+function do_await_full_synchronization() {
+    local NODE_ID=${1}
+    local WAIT_TIME_SEC=0
+    log_step "awaiting full synchronization of node=${NODE_ID}…"
+    while [ "$(do_read_lfb_hash ${NODE_ID})" != "$(do_read_lfb_hash '1')" ]; do
+        if [ "$WAIT_TIME_SEC" = "$SYNC_TIMEOUT_SEC" ]; then
+            log "ERROR: Failed to synchronize node-${NODE_ID} in ${SYNC_TIMEOUT_SEC} seconds"
+            exit 1
+        fi
+        WAIT_TIME_SEC=$((WAIT_TIME_SEC + 1))
+        sleep 1.0
+    done
+}
```

Comment:
Do we need this if we already have `check_network_sync`?

Reply:
Yea i debated removing it. I think it could go with the network sync taking its place

Reply:
removed in: https://github.com/CasperLabs/casper-node/pull/959/commits/9b0e9d05529c9c5b7f9239cbccf8e56d89785f0b